### ISSUE TITLE
inaccurate display of last sync after adding a new connection to an existing pipeline

### ISSUE THREAD

**Issue by jakejh** [DalgoT4D/DDP_backend#525]
See attached. the gsheet connection was just created and added to the same pipeline as the other two connections, but it has not been run.
<img width="1739" alt="Screenshot 2024-03-28 at 21 59 11" src="https://github.com/DalgoT4D/DDP_backend/assets/2295314/ab24bfa2-57f9-4b9a-90ff-2f2404f682fb">

**Comment by fatchat** [comment]
Yes this is a bug, it thinks it was run because the pipeline it's now a part of was run

This is a design flaw, will need to figure out a solution

**Comment by jakejh** [comment]
I just noticed a related behavior: the connection below was part of a pipeline that had been running daily for the last month, but I deleted the pipeline today. prior to deleting the pipeline, dalgo said the last sync was 20 hours ago. now, apparently because the pipeline no longer exists, it says the last sync was 10 days ago, which is probably when I last ran a manual sync.

<img width="1416" alt="Screenshot 2024-05-03 at 13 01 51" src="https://github.com/DalgoT4D/DDP_backend/assets/2295314/21786ce0-7012-4020-b4e1-a64b2d349c93">

**Comment by fatchat** [comment]
your guess is correct - dalgo looks through the list of manual syncs and pipeline runs for pipelines this connection is a part of (these lists are managed separately as distinct prefect deployments)

the change we need to implement is to look directly at the airbyte history rather than the prefect history

**Comment by Ishankoradia** [comment]
I see 3 ways to do this

1) For each connection, we go & fetch the history of syncs & pull the latest one. This might make our get connections api even slower. There is no api to fetch syncs of multiple connections together in airbyte.

2) Setup a new api to fetch the last sync of a single connection_id. On the frontend we refactor a bit, to make sure other information loads while this can be lazy loaded

3) We could setup webhooks in airbyte (this will have to be worksapce by workspace) & start storing each conneciton sync job in our db and use it to fetch the last sync run. 

I am leaning towards 2).

@fatchat what do you think ?

**Comment by fatchat** [comment]
we want the last sync time right
for sid's issue we want stream-level stats
i'd like to pull all this information into a table in our db for analysis i.e. we don't only fetch this info when a user wants to see it on the dashboard

we could populate this by
- running a nightly job 
- adding a task to the end of sync_connection in prefect_airbyte 

the dashboard would attempt to find the information in this table, failing which it will trigger a fetch from airbyte

@Ishankoradia how does this sound

**Comment by Ishankoradia** [comment]
I like the sound of it.

Can we do this `adding a task to the end of sync_connection in prefect_airbyte` by not creating a new task but instead just post a webhook from prefect_airbyte (at end of the sync) to dalgo that starts a celery job ?

And instead do it in our webhook by triggering a celery task.

**Comment by Ishankoradia** [comment]
We might as well start storing the reset/clear jobs in our db. 

So that means i will need to hit the webhook in multiple places in prefect_airbyte (different flows)
